scanner/scanner.go

```diff
@@ -203,12 +234,181 @@ func scanNot(s *Scanner) (token.Token, []byte) {
 	return token.Not, nil
 }
 
+func isInsertPrefix(c byte) bool {
+	return c == '@' || c == '$' || c == '{'
+}
+
 func scanDoubleQuote(s *Scanner) (token.Token, []byte) {
-	for s.char != '"' && s.err == nil {
+	t := token.String
+	next, rOffset := decodeEscapes(s, '"')
+	switch {
+	case isInsertPrefix(next):
+		t = token.StringPart
+		s.pushCtx(stateInDoubleQoutes)
+	case next == '"':
+		s.next()
+	}
+	off := s.offset - rOffset
+	if off > s.begin+1 {
+		off--
+	}
+	return t, s.src[s.begin+1 : off]
+}
+
+func decodeEscapes(s *Scanner, term byte) (byte, int) {
```

Comment:
__[gometalinter]__
* [gocyclo] cyclomatic complexity 12 of function decodeEscapes() is high (> 10)

<!-- Comment by [SideCI](https://sideci.com) -->